File: py-tests/pages/home_page.py
from playwright.sync_api import Page


class HomePage:
    def __init__(self, page: Page) -> None:
        self.page = page

    def open(self, url: str) -> "HomePage":
        self.page.goto(url)
        return self

    def get_title(self) -> str:
        return self.page.title()

    def get_url(self) -> str:
        return self.page.url

    def url_contains(self, text: str) -> bool:
        return text in self.get_url()

File: ts-tests/src/pages/home-page.ts
import { Page } from '@playwright/test';

export class HomePage {
  readonly page: Page;

  constructor(page: Page) {
    this.page = page;
  }

  async open(url: string) {
    await this.page.goto(url);
  }

  async getTitle() {
    return this.page.title();
  }

  async getUrl() {
    return this.page.url();
  }

  async urlContains(text: string) {
    const url = await this.getUrl();
    return url.includes(text);
  }
}
